client/src/main/java/io/pravega/client/tables/KeyUpdateListener.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables;
+
+import java.util.function.Consumer;
+
+/**
+ * Defines a Listener for Key Updates.
+ *
+ * @param <KeyT>   Key Type.
+ * @param <ValueT> Value Type.
+ */
+public interface KeyUpdateListener<KeyT, ValueT> extends AutoCloseable {
```

Comment:
I rewrote this API. It allows you to register lambdas for updates, removals and closes (seal/delete/merge/ etc).

As for the lifecycle, the object itself should stay registered until you hit `close`. It will have a corresponding server-side listener that will push updates, but I will leave it to the implementation to figure out how to handle the ugly cases of disconnects and other problems.